src/main/java/org/stellar/sdk/Transaction.java

```diff
@@ -228,9 +248,12 @@ public static Transaction fromV0EnvelopeXdr(TransactionV0Envelope envelope, Netw
 
   public static Transaction fromV1EnvelopeXdr(AccountConverter accountConverter, TransactionV1Envelope envelope, Network network) {
     int mFee = envelope.getTx().getFee().getUint32();
+    TimeBounds mTimeBounds = null;
     Long mSequenceNumber = envelope.getTx().getSeqNum().getSequenceNumber().getInt64();
     Memo mMemo = Memo.fromXdr(envelope.getTx().getMemo());
-    TimeBounds mTimeBounds = TimeBounds.fromXdr(envelope.getTx().getTimeBounds());
+    if (envelope.getTx().getCond().getDiscriminant().equals(PreconditionType.PRECOND_TIME)) {
+      mTimeBounds = TimeBounds.fromXdr(envelope.getTx().getCond().getTimeBounds());
+    }
```

Comment:
necessary but not sufficient: there can also be timebounds on the `PRECOND_V2` type. maybe a helper is in order for a general way to retrieve timebounds given a tx?

Reply:
yes, I'll use that suggestion, will be in the next follow-on PR that adds the CAP21 support on those new conditions, I included all the xdr schema changes up front in this pr, but only did the cap40 changes above first, this part ideally was just trying to refactor to maintain present compile/test until then.

Reply:
a way to do it here actually would be to say instead `.notEquals(PreconditionType.PRECOND_NONE)`, since both other forms have a timebound condition :thinking: